mmv1/third_party/terraform/services/tpuv2/resource_tpu_v2_vm_test.go.erb

```diff
@@ -35,63 +30,134 @@ func TestAccTpuV2Vm_update(t *testing.T) {
 				ResourceName:            "google_tpu_v2_vm.tpu",
 				ImportState:             true,
 				ImportStateVerify:       true,
-				ImportStateVerifyIgnore: []string{"zone"},
+				ImportStateVerifyIgnore: []string{"labels", "terraform_labels", "zone"},
+			},
+			{
+				Config: testAccTpuV2Vm_update(context, true),
+			},
+			{
+				ResourceName:            "google_tpu_v2_vm.tpu",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"labels", "terraform_labels", "zone"},
 			},
 			{
-				Config: testAccTpuV2Vm_update(updatedContext),
+				Config: testAccTpuV2Vm_update(context, false),
 			},
 			{
 				ResourceName:            "google_tpu_v2_vm.tpu",
 				ImportState:             true,
 				ImportStateVerify:       true,
-				ImportStateVerifyIgnore: []string{"zone"},
+				ImportStateVerifyIgnore: []string{"labels", "terraform_labels", "zone"},
 			},
 		},
 	})
 }
 
 func testAccTpuV2Vm_full(context map[string]interface{}) string {
 	return acctest.Nprintf(`
-data "google_tpu_v2_runtime_versions" "available" {
+resource "google_tpu_v2_vm" "tpu" {
   provider = google-beta
+
+  name        = "tf-test-test-tpu-%{random_suffix}"
+  zone        = "us-central1-c"
+  description = "Text description of the TPU."
+
+  runtime_version  = "tpu-vm-tf-2.13.0"
+  accelerator_type = "v2-8"
+
+  scheduling_config {
+    preemptible = true
+  }
+
+  data_disks {
+    source_disk = google_compute_disk.disk.id
+    mode        = "READ_ONLY"
+  }
+
+  labels = {
+    foo = "bar"
+  }
+
+  metadata = {
+    foo = "bar"
+  }
+
+  tags = ["foo"]
+
+  lifecycle {
+    prevent_destroy = true
+  }
 }
 
-data "google_tpu_v2_accelerator_types" "available" {
+resource "google_compute_disk" "disk" {
   provider = google-beta
+
+  name  = "tf-test-tpu-%{random_suffix}"
+  image = "debian-cloud/debian-11"
+  size  = 10
+  type  = "pd-ssd"
+  zone  = "us-central1-c"
 }
+`, context)
+}
+
+func testAccTpuV2Vm_update(context map[string]interface{}, preventDestroy bool) string {
+	context["prevent_destroy"] = strconv.FormatBool(preventDestroy)
 
+	return acctest.Nprintf(`
 resource "google_tpu_v2_vm" "tpu" {
   provider = google-beta
 
-  name = "tf-test-test-tpu%{random_suffix}"
-  zone = "us-central1-c"
-  description = "%{description}"
+  name        = "tf-test-test-tpu-%{random_suffix}"
+  zone        = "us-central1-c"
+  description = "Text description of the TPU updated."
 
   runtime_version  = "tpu-vm-tf-2.13.0"
   accelerator_type = "v2-8"
-}
-`, context)
-}
 
-func testAccTpuV2Vm_update(context map[string]interface{}) string {
-	return acctest.Nprintf(`
-data "google_tpu_v2_runtime_versions" "available" {
-  provider = google-beta
+  scheduling_config {
+    preemptible = true
+  }
+
+  data_disks {
```

Comment:
Can we add a disk present in both configs that we flip from one `mode` to another?

Reply:
I initially tried that but wasn't able to get a test working with the `READ_WRITE` value on a disk I created:
```
Error 400: Cloud TPU received a bad request. Requested data disk is invalid: disk "projects/terraform-dev-ryanoaks/zones/us-central1-a/disks/tf-test-tpu2-gme8kmr81r" is invalid: generic::invalid_argument: User specified attached disks that are shared across a pod must be in 'READ ONLY' mode, but disk was in mode 
```

Reply:
Weird! We should be working with a single node here if I understand the API correctly.

Reply:
This actually worked when I tried again, so it is included now. I suspect something about my config was making the `READ_WRITE` option unavailable, but not sure what specifically 🤷